### ISSUE TITLE
Getting error to execute nexus file; 

### ISSUE THREAD

**Issue by zillurbmb51** [NBISweden/MrBayes#66]
I was trying to run mrbayes with the following nexus file. `#NEXUS
[ Title fasta file]
begin taxa;
   dimensions ntax= 43;
   taxlabels
      BbigeminaBOND          
      BbovisT2Bo             
      BmicrotiRI             
      Candersoni30847        
      Ccayetanensis          
      CfelisWinnie           
      ChominisTU502          
      CmurisRN66             
      CparvumIowaII          
      Cubiquitum39726        
      CveliaCCMP2878         
      EacervulinaHoughton    
      EfalciformisB          
      EmaximaW               
      EmitisH                
      EnecatrixH             
      EtenellaH              
      GniphandrodesUnknown   
      HhammondiH             
      NcaninumL              
      PbergheiANKA           
      Pchabaudichabaudi      
      PcoatneyiHackeri       
      PcynomolgiB            
      Pfalciparum3D7         
      PfragileNilgiri        
      PgaboniSY75            
      Pgallinaceum8A         
      PinuiSanAntonio1       
      PknowlesiH             
      PmalariaeUG01          
      PovalecurtisiGH01      
      PreichenowiCDC         
      PrelictumSGS1          
      PvinckeipetteriCR      
      PvivaxSal1             
      Pyoeliiyoelii17X       
      SneuronaSN3            
      TannulataAnkara        
      TgondiiVEG             
      TorientalisShintoku    
      TparvaMuguga           
      VbrassicaformisCCMP3155
;
end;
begin characters;
   dimensions nchar= 615642;
   format missing=? gap=- matchchar=. datatype=protein interleave=yes;
   matrix

[!Domain=Data;]
end;
begin mrbayes;
lset applyto = (all) nucmodel=Protein nst=6 rates=gamma covarion=no; [JTTDCMut+F+R5]
unlink statefreq=(all) revmat=(all) shape=(all) pinvar=(all);
prset applyto=(all) ratepr=variable;
mcmc ngen=1000000 temp=0.1 nchain=4 samplefreq=1000 printfr=1000 nruns=4 savebrlens=yes;
end;
sumt filename=test_cl_fa.nex BurninFrac=0.1 Nruns=4 contype=allcompat conformat=Figtree;
`
But getting the following error. What does it mean? What should I do? Thanks in advance.

 `   Executing file "test_cl_fa.nex"
   UNIX line termination
   Longest line length = 105
   Parsing file
   Expecting NEXUS formatted file
   Reading taxa block
      Allocated taxon set
      Defining new set of 43 taxa
   Exiting taxa block
   Reading characters block
      Allocated matrix
      Defining new character matrix with 615642 characters
      Missing data coded as ?
      Gaps coded as -
      Matching characters coded as .
      Data is Protein
      Data matrix is interleaved
      Taxon  1 -> BbigeminaBOND
      Taxon  2 -> BbovisT2Bo
      Taxon  3 -> BmicrotiRI
      Taxon  4 -> Candersoni30847
      Taxon  5 -> Ccayetanensis
      Taxon  6 -> CfelisWinnie
      Taxon  7 -> ChominisTU502
      Taxon  8 -> CmurisRN66
      Taxon  9 -> CparvumIowaII
      Taxon 10 -> Cubiquitum39726
      Taxon 11 -> CveliaCCMP2878
      Taxon 12 -> EacervulinaHoughton
      Taxon 13 -> EfalciformisB
      Taxon 14 -> EmaximaW
      Taxon 15 -> EmitisH
      Taxon 16 -> EnecatrixH
      Taxon 17 -> EtenellaH
      Taxon 18 -> GniphandrodesUnknown
      Taxon 19 -> HhammondiH
      Taxon 20 -> NcaninumL
      Taxon 21 -> PbergheiANKA
      Taxon 22 -> Pchabaudichabaudi
      Taxon 23 -> PcoatneyiHackeri
      Taxon 24 -> PcynomolgiB
      Taxon 25 -> Pfalciparum3D7
      Taxon 26 -> PfragileNilgiri
      Taxon 27 -> PgaboniSY75
      Taxon 28 -> Pgallinaceum8A
      Taxon 29 -> PinuiSanAntonio1
      Taxon 30 -> PknowlesiH
      Taxon 31 -> PmalariaeUG01
      Taxon 32 -> PovalecurtisiGH01
      Taxon 33 -> PreichenowiCDC
      Taxon 34 -> PrelictumSGS1
      Taxon 35 -> PvinckeipetteriCR
      Taxon 36 -> PvivaxSal1
      Taxon 37 -> Pyoeliiyoelii17X
      Taxon 38 -> SneuronaSN3
      Taxon 39 -> TannulataAnkara
      Taxon 40 -> TgondiiVEG
      Taxon 41 -> TorientalisShintoku
      Taxon 42 -> TparvaMuguga
      Taxon 43 -> VbrassicaformisCCMP3155
      Unrecognized Protein character '*'
      Error while reading character position 8683 (charCode -1)
      Deleting previously defined characters
      Deleting previously defined taxa
      Error when setting parameter "MatrixInfo" (2)
      The error occurred when reading char. 67-67 on line 4829
         in the file 'test_cl_fa.nex'

   Returning execution to command line ...

   Error in command "Execute"
   Will exit with signal 1 (error) because quitonerror is set to yes
   If you want control to be returned to the command line on error,
   use 'mb -i <filename>' (i is for interactive) or use 'set quitonerror=no'`

Best Regards
Zillur

**Comment by nylander** [comment]
Dear Zillur,

You get a error message saying:

`Unrecognized Protein character '*'
Error while reading character position 8683 (charCode -1)`

This means that you have a character (`*`) in your data matrix that is not a letter of the IUPAC alphabet for amino acids. This character (and any other similar), needs to be removed. My guess is that you have included a symbol (`*`) for a stop codon?

/J

**Comment by zillurbmb51** [comment]
Thank you very much for quick reply. The problem was solved by replacing "*" with ".". Is it OK? I have a new problem now. Can you give me some hint about this? Thanks in advance.

Best regards
Zillur
`  Successfully read matrix
      Setting default partition (does not divide up characters)
      Setting model defaults
      Seed (for generating default start values) = 1529454898
      Setting output file names to "test_cl_fa_2.nex.run<i>.<p|t>"
   Exiting characters block
   Reading mrbayes block
      Nst =1 unchanged for partition 1 because dataType is not DNA or RNA
      Setting Rates to Gamma for partition 1
      Setting Covarion to No for partition 1
      Successfully set likelihood model parameters
      Unlinking
      Setting Ratepr to Variable [Dirichlet(..,1,..)] for partition 1
      Successfully set prior model parameters
      Setting number of generations to 1000000
      Setting heating parameter to 0.100000
      Setting number of chains to 4
      Setting sample frequency to 1000
      Setting print frequency to 1000
      WARNING: Reallocation of zero size attempted. This is probably a bug. Problems may follow.
      WARNING: Reallocation of zero size attempted. This is probably a bug. Problems may follow.
      Setting number of runs to 4
      Setting chain output file names to "test_cl_fa_2.nex.run<i>.<p/t>"
      Running Markov chain
      MCMC stamp = 0397796218
      Seed = 1881107852
      Swapseed = 1529454898
      Model settings:

         Data not partitioned --
            Datatype  = Protein
            Aamodel   = Poisson
                        Substitution rates are fixed to be equal
            Covarion  = No
            # States  = 20
                        State frequencies are fixed to be equal
            Rates     = Gamma
                        The distribution is approximated using 4 categories.
                        Shape parameter is exponentially
                        distributed with parameter (1.00).

      Active parameters: 

         Parameters
         ---------------------
         Statefreq           1
         Shape               2
         Ratemultiplier      3
         Topology            4
         Brlens              5
         ---------------------

         1 --  Parameter  = Pi
               Type       = Stationary state frequencies
               Prior      = Fixed (equal frequencies)

         2 --  Parameter  = Alpha
               Type       = Shape of scaled gamma distribution of site rates
               Prior      = Exponential(1.00)

         3 --  Parameter  = Ratemultiplier
               Type       = Partition-specific rate multiplier
               Prior      = Dirichlet(1.00)

         4 --  Parameter  = Tau
               Type       = Topology
               Prior      = All topologies equally probable a priori
               Subparam.  = V

         5 --  Parameter  = V
               Type       = Branch lengths
               Prior      = Unconstrained:GammaDir(1.0,0.1000,1.0,1.0)



      The MCMC sampler will use the following moves:
         With prob.  Chain will use move
            1.90 %   Multiplier(Alpha)
            1.43 %   Dirichlet(Ratemultiplier)
            1.43 %   Slider(Ratemultiplier)
            9.52 %   ExtSPR(Tau,V)
            9.52 %   ExtTBR(Tau,V)
            9.52 %   NNI(Tau,V)
            9.52 %   ParsSPR(Tau,V)
           38.10 %   Multiplier(V)
           13.33 %   Nodeslider(V)
            5.71 %   TLMultiplier(V)

      Division 1 has 481760 unique site patterns
      Initializing conditional likelihoods
      Using standard SSE likelihood calculator for division 1 (single-precision)
Killed
You have new mail in /var/spool/mail/zillur
`
Every time I run mb I am getting this. Thank for your help.

**Comment by nylander** [comment]
Sorry, don't know the exact reason. I suspect memory requirement, however. You have a large data matrix. Monitor the use of RAM on your system.
/J

**Comment by siddharthk02** [comment]
I am also having the same error
 Executing file "MrBayes.nexus"
   DOS line termination
   Longest line length = 1187
   Parsing file
   Expecting NEXUS formatted file
   Dimensions can only be defined in a data, characters or taxa block
   Error in command "Dimensions"
   The error occurred when reading char. 32-32 on line 2
      in the file 'MrBayes.nexus'

   Returning execution to command line ...

   Error in command "Execute"
How do I proceed???

**Comment by Jorethadb** [comment]
I had the same problem:

Unrecognized DNA/RNA character type 
Error while reading character position (charCode 1)
Deleting previously defined characters
Deleting previously defined taxa
Error when setting parameter "MatrixInfo" (2)

After struggling for a while I realized that when I converted my sequences from a .txt file to a .fasta file, there were grey "stop blocks" added to the end of all my sequences. In this .fasta file, I had longer and shorter sequences, and when I trimmed the longer sequences, deleting the grey blocks, I didn't look at the shorter sequences (they still had the grey blocks). When I exported the .fasta file to a .nex file, the grey blocks translated to "spaces" within the .nex file. The spaces stopped the program and gave the above error. I had to edit the .nex file by deleting the "spaces" and adding a "-". This finally solved my problem.

Hope this helps someone even if the comment is closed.
